cql3/statements/alter_table_statement.cc

```diff
@@ -388,7 +393,17 @@ future<shared_ptr<cql_transport::event::schema_change>> alter_table_statement::a
         break;
     }
 
-    return service::get_local_migration_manager().announce_column_family_update(cfm.build(), false, std::move(view_updates), is_local_only).then([this] {
+    auto f = service::get_local_migration_manager().announce_column_family_update(cfm.build(), false, std::move(view_updates), is_local_only);
+    if (create_cdc) {
+        f = f.then([&proxy, schema = std::move(schema)] {
+            return cdc::setup(cdc::db_context::builder(proxy).build(), schema);
+        });
+    } else if (delete_cdc) {
+        f = f.then([&proxy, schema = std::move(schema)] {
+            return cdc::remove(cdc::db_context::builder(proxy).build(), schema->ks_name(), schema->cf_name());
+        });
+    }
+    return f.then([this] {
         using namespace cql_transport;
```

Comment:
This is totally broken if we crash between creating the table and creating the cdc log.

However, there's no easy way to fix it until we get transactional schema, so let's live with it for now.

Reply:
I guess I could reverse the order so that I first setup CDC and only after that announce the schema change. It would be a bit safer.